### DISCUSSION TITLE
On Mac, move todo App to current space with keystroke

### DISCUSSION THREAD

**Discussion by juicero-chief-juice-officer** [rxhanson/Rectangle#1279] [Ideas]
I keep several "Spaces", but my todo app (TickTick) only allows one window at a time. 

I'd love a Rectangle keystroke that pulls the designated todo app/window onto the active/current space and adjusts the other windows accordingly (eg if i designate 400 out of 4000 pixels for the to do app, on entering the keystroke apps occupying the left 2000 pixels now occupy the left 1800 pixels, etc.) 

Mac Spaces is so limited that any enhancements Rectangle can offer are very welcome.  (Edited to add - I see [Space Capsule](https://spacecapsule.app/) now and will check it out.)

**Comment by rxhanson** [comment]
Thanks for the feedback. There's no supported API in macOS to do this. The closest I have in any of my apps is the ability to move a window to the next/prev space, in Rectangle Pro via keyboard shortcut or Multitouch as a configured action.

**Reply by ilyagr** [reply]
An alternative would be to make the TODO app show up on "All Desktops" (like you get if you right-click on an app and go to Options) and, perhaps, make it be always-on-top. Is there no API for that either?

**Reply by rxhanson** [reply]
Unfortunately, there is no API for that either. An app can make its own windows show up on all desktops, but it cannot make windows for another app show up on all desktops.

**Reply by ilyagr** [reply]
Oh well. Thanks for looking into it, and for making Rectangle!

**Reply by glamout** [reply]
As we know, while dragging a window, press key combo to move window to numbered space:
Control + 1, Control + 2, Control + 3...
I am wondering if can refer this to add new  keystroke in the rectangle, thanks.

**Reply by glamout** [reply]
FYI, here is a app can do it
https://www.irradiatedsoftware.com/sizeup/

**Reply by rxhanson** [reply]
@glamout thanks for the feedback. Next/prev space is included in Rectangle Pro, but not in Rectangle. I'll keep the control+number idea in mind for future updates.

**Reply by glamout** [reply]
Great, thanks.
